src/lib/storage/segment_encoding_utils.cpp

```diff
@@ -51,11 +51,24 @@ std::shared_ptr<BaseEncodedSegment> encode_and_compress_segment(const std::share
   return encoder->encode(segment, data_type);
 }
 
-/**
- * @brief Returns the vector compression type for a given compressed vector type.
- *
- * For the difference of the two, please take a look at compressed_vector_type.hpp.
- */
+SegmentEncodingSpec get_segment_encoding_spec(const std::shared_ptr<const BaseSegment>& segment) {
+  Assert(!std::dynamic_pointer_cast<const ReferenceSegment>(segment), "Reference segments cannot be encoded.");
+
+  if (std::dynamic_pointer_cast<const BaseValueSegment>(segment)) {
+    return SegmentEncodingSpec{EncodingType::Unencoded};
+  }
+
+  if (const auto encoded_segment = std::dynamic_pointer_cast<const BaseEncodedSegment>(segment)) {
+    std::optional<VectorCompressionType> vector_compression;
+    if (encoded_segment->compressed_vector_type()) {
+      vector_compression = parent_vector_compression_type(*encoded_segment->compressed_vector_type());
+    }
+    return SegmentEncodingSpec{encoded_segment->encoding_type(), vector_compression};
+  }
+
+  Fail("Unexpected segment encoding found.");
```

Comment:
Guard clause?

Reply:
Woud require to check for a couple of possible dynamic_casts, wouldn't it? I don't have an idea how to create a non-ugly guard here. :-/

Reply:
Ah right, that's a chain of ifs. Never mind.